src/tile2net/raster/tile.py

```diff
@@ -306,17 +321,23 @@ def map_features(self, src_img, img_array=True):
 
     def get_region(self, gdf: gpd.GeoDataFrame, spatial_index, crs=3857):
         """
-        clips the overlapping region between the dataframe and tile extent
-        Args:
-            tile: the tile object to overlay
-            gdf: the
-
-        Returns:
+        Clips the overlapping region between a given GeoDataframe and a :class:`GeoDataFrame.sindex` when creating masks. 
+        See the [Geopandas documentation](https://geopandas.org/en/stable/docs/reference/api/geopandas.GeoDataFrame.sindex.html) for more
+        information on spatial indices. 
 
         Parameters
         ----------
-        spatial_index
+        gdf: :class:`GeoDataFrame`
+            The :class:`GeoDataFrame` to clip
+        spatial_index: :class:`GeoDataFrame.sindex`
+            The spatial index to clip the :class:`GeoDataFrame` against
+        crs: int
+            The desired coordinate reference system to save the network polygon with.
 
+        Returns
+        -------
+        :class:`GeoDataFrame` | int
```

Comment:
```suggestion
        :class:`GeoDataFrame` or int
```
numpy docstring style uses "or" instead or "|":
```    Returns
    -------
    out : ndarray or scalar
        Output array, element-wise comparison of `x1` and `x2`.
        Typically of type bool, unless ``dtype=object`` is passed.
        This is a scalar if both `x1` and `x2` are scalars.
```